lib/remote-proxy.js

```diff
@@ -1,69 +1,64 @@
 'use strict';
 
-const events = require('events');
-const util = require('util');
+const EventEmitter = require('events').EventEmitter;
 
 // Remote API proxy object class. It wraps remote methods so that they look
 // like regular local methods and acts like a remote event emitter.
-//   connection - JSTP connection to use
-//   interfaceName - name of an interface that is proxied
-//   methods - array of method names (optional)
 //
-function RemoteProxy(connection, interfaceName, methods = []) {
-  events.EventEmitter.call(this);
+class RemoteProxy extends EventEmitter {
+  //   connection - JSTP connection to use
+  //   interfaceName - name of an interface that is proxied
+  //   methods - array of method names (optional)
+  //
+  constructor(connection, interfaceName, methods = []) {
+    super();
 
-  this._connection = connection;
-  this._interfaceName = interfaceName;
+    this._connection = connection;
+    this._interfaceName = interfaceName;
 
-  for (let i = 0; i < methods.length; i++) {
-    RemoteProxy.wrapRemoteMethod(this, methods[i]);
+    for (let i = 0; i < methods.length; i++) {
+      this._wrapRemoteMethod(methods[i]);
+    }
   }
-}
 
-util.inherits(RemoteProxy, events.EventEmitter);
+  // Emit an event.
+  //   eventName - name of an event
+  //   eventArgs - event arguments
+  //
+  emit(eventName, ...eventArgs) {
+    this._connection.emitRemoteEvent(this._interfaceName, eventName, eventArgs);
+    this._emitLocal(eventName, eventArgs);
+  }
 
-// Emit an event.
-//   eventName - name of an event
-//   eventArgs - event arguments
-//
-RemoteProxy.prototype.emit = function(eventName, ...eventArgs) {
-  this._connection.emitRemoteEvent(this._interfaceName, eventName, eventArgs);
-  this._emitLocal(eventName, eventArgs);
-};
+  // Emit local event.
+  //   eventName - name of an event
+  //   eventArgs - array of event arguments
+  //
+  _emitLocal(eventName, eventArgs = []) {
+    super.emit(eventName, ...eventArgs);
+  }
 
-// Emit local event.
-//   eventName - name of an event
-//   eventArgs - array of event arguments
-//
-RemoteProxy.prototype._emitLocal = function(eventName, eventArgs = []) {
-  events.EventEmitter.prototype.emit.call(this, eventName, ...eventArgs);
-};
 
-// Create a method in a remote proxy that will call the corresponding remote
-// method. This is implemented as a static method rather than an instance
```

Comment:
> This is implemented as a static method rather than an instance	method so that it will not be rewritten by a remote API method with the same	name.

Why did you change it?
I think that reasoning here is clear and it should still be a static method after refactoring.